src/docs/arch-schema.dox

```diff
@@ -1,11 +1,117 @@
+operation:
+
+@plantuml_start{schema_generic.png }
+@startuml{schema_generic.png}
+:A schema operation;
+partition with-schema-lock {
+    :perform operation on underlying data-object;
+    :update metadata-file;
+    :checkpoint and sync metadata;
+}
+stop
+@enduml
+@plantuml_end
+
+Apart from the schema API, the schema lock is necessary for many other
+operations in WiredTiger including the following "heavy" database modifications:
+ - The schema lock wraps `checkpoint prepare` to avoid any tables being created or dropped
+ during this phase. See @ref arch-checkpoint "Checkpoint" for details.
+ - `Rollback to stable` operation acquires the schema lock to make sure no schema changes are
+ done during this complex process. @ref arch-rts "Rollback to Stable" has more information on the
+ operation.
+ - A `backup cursor` also holds the schema lock because it must guarantee a consistent view
+ of what files and tables exist while it is being used, so it prevents any tables or files being
+ created or dropped during that time. See @ref backup and @ref arch_cursor_backup for
+ more information.
+
+All the schema operations listed below perform multi-step metadata modifications. Although they
+are \c non-transactional, the schema code tracks the metadata changes and performs the file and
+metadata operations in a specific order to provide recovery in the case of a crash.
+
+@section schema_api Schema Manipulation
+
+All schema manipulations are done in the context of WT_SESSION. All the methods below,
+except WT_SESSION::create and WT_SESSION::truncate, require exclusive access to the specified
+data source(s). If any cursors are open with the specified name(s) or a data source is otherwise
+in use, the call will fail and return EBUSY.
+
+@subsection schema_create Create
+
+The create schema operation is responsible for creating the underlying data
+objects on the filesystem and then creating required entries in the metadata. The API for this
+operation is WT_SESSION::create.
+
+@subsection schema_drop Drop
+
+WT_SESSION::drop operation drops the specified \c uri. The method will delete all related files
+and metadata entries. It is possible to keep the underlying files by specifying
+\c "remove_files=false" in the config string.
+
+@subsection schema_rename Rename
+
+WT_SESSION::rename schema operation renames the underlying data objects on the filesystem and
+updates the metadata accordingly.
+
+@subsection schema_alter Alter
+
+WT_SESSION::alter allows modification of some table settings after creation.
+
+@subsection schema_truncate Truncate
+
+WT_SESSION::truncate truncates a file, table, cursor range, or backup cursor. If \c start and
+\c stop cursors are not specified all the data stored in the \c uri will be wiped out. When a range
+truncate is in progress, and another transaction inserts a key into that range, the behavior
+is not well defined. It is best to avoid this type of situations.
```

Comment:
We can link this section to where we talk about truncate in the btree/dhandle arch guide page. That page is not checked-in yet. Here is the portion on truncate:
https://github.com/wiredtiger/wiredtiger/pull/6904/files#diff-839bc7e4663c1dc228631c2dbf82594f0b3ee09f1bc8f55c6d1be4d6c1f4692cR45

Reply:
I have just merged in my architecture guide, so it's possible to link it directly if you want. It should be on arch-btree as I have separated these pages.

Reply:
Thank you @sulabhM and @mm-ng ! I've added the link.